corpus/pages/views.py

```diff
@@ -41,3 +45,126 @@ def sig(request, sig_name):
+
+    ]
+
+    diode_core_posts = [
+        "Diode Chair",
+        "SPS Chair",
+        "SPS Vice Chair",
+        "SPS Secretary",
+        "CAS Chair",
+        "CAS Vice Chair",
+        "CAS Secretary",
+        "RAS Chair",
+        "RAS Secretary",
+    ]
+
+    piston_core_posts = [
+        "Piston Chair",
+        "Piston Vice Chair",
+        "Piston Secretary",
+        "Piston Project Head",
+        "IAS Chair",
+        "IAS Secretary",
+    ]
+
+    ieee_core = Core.objects.filter(post__in=ieee_core_posts).order_by(
+        Case(
+            When(post="Convenor", then=Value(1)),
+            When(post="Chairperson", then=Value(2)),
+            When(post="Vice Chairperson", then=Value(3)),
+            When(post="Secretary", then=Value(4)),
+            When(post="Joint Secretary", then=Value(5)),
+            When(post="Treasurer(Branch)", then=Value(6)),
+            When(post="Treasurer(Institute)", then=Value(7)),
+            When(post="Webmaster", then=Value(8)),
+            When(post="Media Lead", then=Value(9)),
+            When(post="Outreach Lead", then=Value(10)),
+            When(post="Envision Lead", then=Value(11)),
+            When(post="Labs Lead", then=Value(12)),
+            When(post="WiE Chair", then=Value(13)),
+        )
+    )
+
+    compsoc_core = Core.objects.filter(post__in=compsoc_core_posts).order_by(
+        Case(
+            When(post="CompSoc Chair", then=Value(1)),
+            When(post="CompSoc Vice Chair", then=Value(2)),
+            When(post="CompSoc Secretary", then=Value(3)),
+            When(post="CompSoc Project Head", then=Value(4)),
+            When(post="CompSoc Project Coordinator", then=Value(5)),
+            When(post="CIS Chair", then=Value(6)),
+            When(post="CIS Secretary", then=Value(7)),
+            When(post="CIS Project Head", then=Value(8)),
+            When(post="Webmaster", then=Value(9)),
+        )
+    )
+
+    diode_core = Core.objects.filter(post__in=diode_core_posts).order_by(
+        Case(
+            When(post="Diode Chair", then=Value(1)),
+            When(post="SPS Chair", then=Value(2)),
+            When(post="SPS Vice Chair", then=Value(3)),
+            When(post="SPS Secretary", then=Value(4)),
+            When(post="CAS Chair", then=Value(5)),
+            When(post="CAS Vice Chair", then=Value(6)),
+            When(post="CAS Secretary", then=Value(7)),
+            When(post="RAS Chair", then=Value(8)),
+            When(post="RAS Secretary", then=Value(9)),
+        )
+    )
+
+    piston_core = Core.objects.filter(post__in=piston_core_posts).order_by(
+        Case(
+            When(post="Piston Chair", then=Value(1)),
+            When(post="Piston Vice Chair", then=Value(2)),
+            When(post="Piston Secretary", then=Value(3)),
+            When(post="Piston Project Head", then=Value(4)),
+            When(post="IAS Chair", then=Value(5)),
+            When(post="IAS Secretary", then=Value(6)),
+        )
+    ) 
```

Comment:
Is it not possible to define a sort function that does this for you, instead of using this switch case logic?
In the sort function you can define this precedence order, and have the list sorted by the precedence order you define.

Here's a pretty old but still relevant [StackOverflow thread](https://stackoverflow.com/questions/883575/custom-ordering-in-django) for more context.